File: Chapter 2/selection.py
def FindSmallest(arr):
    smallest = arr[0];
    smallest_index=0;
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index


def Selection_Sort(arr):
    newArr = [];
    for i in range(len(arr)):
        smallest = FindSmallest(arr)
        newArr.append(arr.pop(smallest))
    return newArr

print(Selection_Sort([5, 3, 6, 2, 10]))

File: Chapter 2/selection.js
const findSmallest = (arr) => {
    smallest = arr[0];
    smallest_index = 0;

    for(let i = 0; i < arr.length; i++){
        if(arr[i] < smallest){
            smallest = arr[i];
            smallest_index = i;
        }    
    }
    return smallest_index;
}

const selectionSort = (arr) => {
    let newArr = [];
    if(!arr.length) {
        return [];
    }
    for(let i = 0; i < arr.length; i++){
        let smallest = arr.splice( findSmallest( arr ), 1 );
        newArr.push(smallest);
        return smallest.concat( selectionSort( arr ) );

    }
    return newArr;
}

console.log(selectionSort([55, 323, 63, 2, 10]));